### PR TITLE
Compute canonical path before adjusting parent path

### LINKED ISSUE #854: page.translations contains unrelated pages
# Bug Report

I tried getting the translated versions of a page through:

```
{% for translation in page.translations %}
    {{ translation.title }} {{ translation.lang }}<br>
{% endfor %}
```

Even though the page has only a single translation, the array contains also contains completely unrelated (untranslated) pages. This makes it impossible to find the correct translation.

## Environment

Zola version: 0.9.0

## Expected Behavior
The `translations` array only has a single entry: The translated page. (Alternatively, it could also contain the default untranslated page for the default language.)

### PR DESCRIPTION
**Pull request by phil-opp**
Fixes #854

### CODE REVIEW

**Comment by phil-opp** [comment]
I think we also need to strip the language tag from the filename when generating the canonical path, otherwise translations are not found. Do you want me to add this change to this PR or do you prefer when I open a new one?

**Comment by Keats** [comment]
I mean that's weird that it was not doing it, I swear I had code to handle that.. :(
Can you add some tests while you're there? At least a regression test with a link to the GH issue.

**Comment by phil-opp** [comment]
> I mean that's weird that it was not doing it, I swear I had code to handle that.. :(

Maybe you mean this code? https://github.com/getzola/zola/blob/6d7c95e8cb6c60336119a9aeb53c1d30f1d5d2df/components/library/src/content/file_info.rs#L114-L143

This line seems to have the same problem:

https://github.com/getzola/zola/blob/6d7c95e8cb6c60336119a9aeb53c1d30f1d5d2df/components/library/src/content/file_info.rs#L138

You're using the adjusted `parent` field to create the canonical path again.

**Comment by phil-opp** [comment]
I pushed a second commit that also fixes the issue in `find_language`.

**Comment by phil-opp** [comment]
I added some basic regression tests for the two commits.

File: components/library/src/content/file_info.rs [hunk 1]
```diff
@@ -254,4 +255,28 @@ mod tests {
...
+    fn correct_canonical_after_find_language() {
+        let mut config = Config::default();
+        config.languages.push(Language { code: String::from("fr"), rss: false, search: false });
+        let mut file = FileInfo::new_page(
+            &Path::new("/home/vincent/code/site/content/posts/tutorials/python/index.fr.md"),
+            &PathBuf::new(),
+        );
+        let res = file.find_language(&config);
+        assert!(res.is_ok());
+        assert_eq!(file.canonical, Path::new("/home/vincent/code/site/content/posts/tutorials/python/index"));
```
Comment (Keats) [inline components/library/src/content/file_info.rs:280]: I think the reason for those bugs is that in my mind the canonical path was `/home/vincent/code/site/content/posts/tutorials/python`.
Anyway, thanks a lot!

**Pull request merged**